sdpsensor.cpp

```diff
@@ -80,7 +80,7 @@ int SDPSensor::readSample()
 
   int16_t dp_raw   = (int16_t)data[0] << 8 | data[1];
   int16_t temp_raw = (int16_t)data[3] << 8 | data[4];
-  int8_t dp_scale  = (int16_t)data[7] << 8 | data[8];
+  uint8_t dp_scale  = (int16_t)data[6] << 8 | data[7];
```

Comment:
Hi @teemune, many thanks for the patch. I'll double check the validity with our SDP team - one thing that does strike me as odd is that we're casting an int16 into a uint8, which basically discards the upper 8 bits (this is not introduced by you, it was already the case) but your change thus now assigns `data[7]` to `dp_scale` and treats it as unsigned, instead of `data[8]` as signed value - `data[8]` is likely the CRC anyway :/

Reply:
agreed, `data[8]` is the CRC so the fixed indices are certainly correct, while the data type should probably be a 16-bit type as well (and was wrong in the first place)

Reply:
@abrauchli looks like the sample code on sensirion.com is using uint16_t for the DP scale factor

Reply:
Hi, I figured I'll make the least amount of changes. data[6] contains no information with such small numbers, so you'll be the best judge if it's better to use a bigger variable or just remove the usage of data[6].

Reply:
I check with our SDP experts. In section 5.5.3 the value is specified as two's complement signed integer, thus the correct type is `int16_t`.

https://www.sensirion.com/fileadmin/user_upload/customers/sensirion/Dokumente/0_Datasheets/Differential_Pressure/Sensirion_Differential_Pressure_Sensors_SDP3x_Digital_Datasheet.pdf p10

Could you thus adapt the PR to reflect this?

```c
int16_t dp_scale  = (int16_t)data[6] << 8 | data[7];
```

Reply:
turns out sec5.5.2 is not clear enough and the people who would know are on vacation.. all in all it is highly unlikely that we'll run into the issue. Given the little practical relevance I'm happy to merge it as signed 16bit value, as suggested above, and repatch later if needed.

Reply:
Updated accordingly to my code. It would be logical as other values are the same format int16 should read correct. 
register value I read:
0000000011110000
with int8 it converts to -16, int16 it converts correctly to 240.

Reply:
The change should now be visible in this pull request too.